### PR TITLE
MAINT: numpy cleanup version bumps: fixes issue #20458

### PR DESCRIPTION
**Pull request by peppedilillo**
The minimum numpy version in `scipy/scipy/__init__.py` has been aligned to the minimum requirement of `pyproject.toml`. A numpy version check has been removed from test `test_moments` in `scipy/stats/tests/test_multivariate.py` because the check looked for a numpy version earlier than the minimum. This pull request does not fully solve #20458. I will come back at it after discussion.

#### Reference issue
Partially solves issue #20458 

#### What does this implement/fix?
This aligns a few numpy versions. Does not align the pytest version.

#### Additional information
I'm open to working on fixing the pytest versions too, as reported in #20458. However, it is not clear to me what is the exact pytest version we should report. Shall we find it ourselves or you already know? Tried grepping for it but didn't find anything useful.

### CODE REVIEW

File: scipy/__init__.py [hunk 1]
```diff
@@ -66,7 +66,7 @@
 
 from scipy._lib import _pep440
 # In maintenance branch, change to np_maxversion N+3 if numpy is at N
-np_minversion = '1.22.4'
+np_minversion = '2.0.0rc1'
```
Comment (tylerjereddy) [inline scipy/__init__.py:69]: This is not correct, it would ban imports of NumPy prior to the 2.x series, which we still allow.

**Review by tylerjereddy — COMMENTED** [review]
This PR is basically a duplicate of https://github.com/scipy/scipy/pull/20459 with an additional problem at the moment.

Also, the point in question, the `pytest` version guard, was already discussed over there at: https://github.com/scipy/scipy/pull/20459#pullrequestreview-2012055393

File: scipy/__init__.py (hunk 1 above)
Comment (peppedilillo) [reply scipy/__init__.py]: I realize this is a serious error, sorry! Also I found PR#20459 just after committing.. I have pushed a fix to the problem you report, and a few changes which should resolve the pytest version guard problem. Sorry for the duplicate and, again, for the unfortunate error!

**Comment by tylerjereddy** [comment]
The CI failures were unrelated, and the diff seems "ok" now, so I've squash-merged it. Note that I've added @JacobOgle as a co-author giving the two active PRs.

**Comment by peppedilillo** [comment]
> The CI failures were unrelated, and the diff seems "ok" now, so I've squash-merged it. Note that I've added @JacobOgle as a co-author given the two active PRs.

Thank you. Great to have added @JacobOgle: i copypasted his numpy version! Thank you too Jakob! 
Bests and catch you soon hopefully. ~peppe

**Pull request merged**